kernel/libc/koslib/readdir.c

```diff
@@ -10,6 +10,13 @@
 #include <string.h>
 #include <kos/fs.h>
 
+#ifdef __STRICT_ANSI__
+/* Newlib doesn't prototype this function in strict standards compliant mode, so
+   we'll do it here. It is still provided either way, but it isn't prototyped if
+   we use -std=c17 (or any other non-gnuXX value). */
+size_t strnlen(const char *, size_t);
```

Comment:
```suggestion
extern size_t strnlen(const char *, size_t);
```
Okay, so this is a Falco-ism, but maybe I can convince you... The default is already `extern` linkage, BUT, I typically explicitly mark the declarations of anything within KOS that it is getting linked from outside--like Newlib, libc, etc--as `extern`, so that someone looking at the coedz knows not to go hunting for the symbol in the KOS codebase and that it's from the outside.

Reply:
I do like the idea as functional commentary *but* that is the role of the giant comment above it, and we kinda wanna have that so that it's easily findable for "newlib", and this is a direct copy+paste of similar ones for other gnu extensions.

I think the thing though might really be to say that we do rely on the gnu extensions and not pretend that we should work in strict standards mode. It's effectively an abuse of newlib to blindly link back into these when we are telling the compiler that they're not allowed. Just not sure if that'd have any implication other than warnings with `-ansi`.

Reply:
I'm going to agree with @QuzarDC here that the comment is good enough. The extern isn't necessary here and without commentary it's actually a bit confusing to see (at least I'd be taken a bit aback by it just browsing the code).